### PR TITLE
fix: parse document arg order adding tests #120

### PR DESCRIPTION
**Pull request by zcstarr**

### CODE REVIEW

**Comment by shanejonas** [comment]
just a nit: the commit message issue ref should be on its own line

```
fix: parse document arg order adding tests 

fixes #120
```

File: package.json [hunk 1]
```diff
@@ -54,9 +54,11 @@
     "@types/json-schema": "^7.0.3",
     "@types/lodash": "^4.14.123",
     "@types/node-fetch": "^2.1.6",
+    "@types/rimraf": "^2.0.2",
```
Comment (BelfordZ) [inline package.json]: we already have fs-extra

File: src/index.test.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  });
+
+  it("should parseOpenRPCDocument from file", async () => {
+    const doc = await parseOpenRPCDocument(testDocPath);
+    expect(_.isEqual(doc, testDoc)).toBe(true);
+  });
+
+  it("should parseOpenRPCDocument from server", async () => {
+    const {port} = testServer.address() as AddressInfo;
+    const doc = await parseOpenRPCDocument(`http://localhost:${port}/download/openrpc.json`);
```
Comment (BelfordZ) [inline src/index.test.ts]: todo: replace this with examples.open-rpc.org when we have it setup

**Comment by BelfordZ** [comment]
The whole server is pretty heavy handed, [especially given we can link to examples on github](https://raw.githubusercontent.com/open-rpc/examples/master/service-descriptions/petstore-openrpc.json)

File: package.json (hunk 1 above)
Comment (zcstarr) [reply package.json]: but rimraf does the whole rm -rf for directories that contain files

File: src/index.test.ts (hunk 2 above)
Comment (zcstarr) [reply src/index.test.ts]: I don't want to do that because its unreliable, to depend on external dependencies for tests. Tests should be self contained

Comment (zcstarr) [reply src/index.test.ts]: and repeatable

**Pull request merged**